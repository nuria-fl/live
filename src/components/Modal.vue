<template lang="html">
  <div v-if="visible" class="Modal">
    <div class="Modal__content">
      <header>
        <slot name="header" />
      </header>
      <section>
        <slot name="body" />
      </section>
      <footer>
        <button v-if="isCloseable" class="Btn" @click="close()">
          {{ closeText }}
        </button>
        <slot name="actions" />
      </footer>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    visible: {
      type: Boolean,
      required: true
    },
    isCloseable: {
      type: Boolean,
      default: false
    },
    closeText: {
      type: String,
      default: 'Ok'
    }
  },
  methods: {
    open() {
      this.$emit('update:visible', true)
    },
    close() {
      this.$emit('update:visible', false)
    }
  }
}
</script>

<style lang="scss">
.Modal {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 9;
  &:before {
    content: '';
    display: block;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    background: rgba(0, 0, 0, 0.6);
  }
  &__content {
    width: 40em;
    max-width: 95%;
    padding: 1em;
    position: absolute;
    top: 40%;
    left: 50%;
    transform: translate(-50%, -50%);
    background: #fff;
    text-align: center;
  }
}
</style>
