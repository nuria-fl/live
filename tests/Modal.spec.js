import { shallow } from 'vue-test-utils'

import Modal from '../src/components/Modal'

describe('Modal', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(Modal, {
      propsData: {
        visible: true,
        isCloseable: true
      }
    })
  })

  test('It should match snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  test('It should emit update event when clicking close button', () => {
    wrapper.find('.Btn:last-child').trigger('click')
    expect(wrapper.emitted('update:visible')).toBeTruthy()

    // weird way to get the event payload
    expect(wrapper.emitted('update:visible')[0][0]).toBe(false)
  })
})
