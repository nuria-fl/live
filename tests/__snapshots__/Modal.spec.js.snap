// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal It should match snapshot 1`] = `
<div
  class="Modal"
>
  <div
    class="Modal__content"
  >
    <header />
     
    <section />
     
    <footer>
      <button
        class="Btn"
      >
        
        Ok
      
      </button>
       
    </footer>
  </div>
</div>
`;
